java-client/src/main/java/org/opensearch/client/json/UnionDeserializer.java

```diff
@@ -169,14 +171,8 @@ public Builder<Union, Kind, Member> addMember(Kind tag, JsonpDeserializer<? exte
             JsonpDeserializer<?> unwrapped = DelegatingDeserializer.unwrap(deserializer);
             if (unwrapped instanceof ObjectDeserializer) {
                 ObjectDeserializer<?> od = (ObjectDeserializer<?>) unwrapped;
-                Set<String> allFields = od.fieldNames();
```

Comment:
Yeah, this was a tricky one :-) May be a bit simpler implementation would be to memorize the already "seen" fields and remove the duplicate as we go (in `addMember`), but that just an idea.